service/handler.go

```diff
@@ -1,8 +1,175 @@
 package service
 
-import "net/http"
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/mux"
+	datadealtypes "github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	"github.com/medibloc/panacea-oracle/crypto"
+	"github.com/medibloc/panacea-oracle/validation"
+	log "github.com/sirupsen/logrus"
+)
+
+type ValidateDataReq struct {
+	ProviderAddress     string `json:"provider_address"`
+	EncryptedDataBase64 string `json:"encrypted_data_base64"`
+	DataHash            string `json:"data_hash"`
+}
 
```

Comment:
How about added validation function like `func (m ValidateDataReq) ValidateBasic() {...}`.
This is a validation of request parameters.